src/Transformer.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\fb_instant_articles\Transformer.
+ */
+
+namespace Drupal\fb_instant_articles;
+
+/**
+ * Encapsulates Drupal-specific logic when using the Transformer class.
+ *
+ * Class Transformer
+ * @package Drupal\fb_instant_articles
+ */
+class Transformer extends \Facebook\InstantArticles\Transformer\Transformer {
+
+  /**
+   * Transformer constructor.
+   *
+   * Adds Drupal hook to allow transformer rules to be added by other modules.
+   * We invoke the hook in a constructor - as opposed to inside transform() or a
+   * new invokeRules() method - because of the way Transformer is normally used.
+   *
+   * @see hook_fb_instant_articles_transformer_rules()
+   *
+   * Note the parent class does not have a constructor, so we do not call
+   * @code parent::__construct() @endcode.
+   */
+  public function __construct() {
+    $rules = module_invoke_all('fb_instant_articles_transformer_rules');
```

Comment:
Makes sense to me as well. I really like the extensibility of this approach 👍

@m4olivei How do you wan to go about implementing in `fb_instant_articles_display`? I'm thinking it needs to be a new PR stacked on top of this one (since this currently removes the loading of rules from json, effectively breaking the functionality of `fb_instant_articles_display`), right?